app/jobs/mirakl_ship_job.rb

```diff
@@ -4,14 +4,30 @@ class MiraklShipJob < ActiveJob::Base
   def perform(store_id, order_id, shipment_id)
     store = Spree::MiraklStore.find(store_id)
     shipment = Spree::Shipment.find(shipment_id)
+
     mirakl_request = SpreeMirakl::Api.new(store)
+    get_order = mirakl_request.get_order(order_id)
+    raise ServiceError.new(["Issue getting #{order_id}"]) unless get_order.success?
+
+    order_data = JSON.parse(get_order.body, symbolize_names: true)[:orders][0]
+    shipping_name = shipment.shipping_method.name.split(' ').try(:first)
```

Comment:
should we move this into the shipment decorator?

Reply:
because it is only ever used here I would leave it here. If we do this in multiple places I would move it to a model method

Reply:
I think it should be a method so it can be overridden if needed. This may not be consistent across spree versions